db/builder.cc

```diff
@@ -47,7 +47,7 @@ TableBuilder* NewTableBuilder(
         int_tbl_prop_collector_factories,
     uint32_t column_family_id, const std::string& column_family_name,
     WritableFileWriter* file, const CompressionType compression_type,
-    const CompressionOptions& compression_opts, int level,
+    uint64_t sample_for_compression, const CompressionOptions& compression_opts, int level,
```

Comment:
I take another look. Seems the argument `sample_for_compression` is redundant, given that `moptions` has a member `sample_for_compression` already.

Reply:
Never mind, I think we can keep it because we sometimes do not sample for certain files/blocks.

Reply:
One question though, what is the intended relationship between `moptions.sample_for_compression` and `sample_for_compression` argument?

Reply:
I think we can keep this as is because it might be desirable to have the flexibility of using different sampling rate for different SST files even of the same column family. I do not have a strong opinion on this. :)